docs/using-injected-bindings/using-injected-bindings.md

```diff
@@ -0,0 +1,165 @@
+---
+sidebar_position: 1
+---
+
+# General Concept
+
+When using Service Binding operator, you'll get a set of binding metadatas injected into your application.
+That injection can be done in two different ways:
+- As files: this is the default behavior,
+- As environment variables: if
+`.spec.bindAsFiles` is set to `false` in the  resource.
+
+## Consuming the Bindings from Applications
+
+The Application Projection section of the spec describes how the bindings are
+projected into the application.  The primary mechanism of projection is through
+files mounted at a specific directory.  The bindings directory path can be
+discovered through `SERVICE_BINDING_ROOT` environment variable setup in your application.  
+The operator must have injected `SERVICE_BINDING_ROOT` environment to all the containers
+where bindings are created.
+
+Within this service binding root directory, there could be multiple bindings
+projected from different Service Bindings.  For example, suppose an application
+requires to connect to a database and cache server. In that case, one Service
+Binding can provide the database, and the other Service Binding can offer
+bindings to the cache server.
+
+Let's take a look at the example given in the spec:
+
+```
+$SERVICE_BINDING_ROOT
+├── account-database
+│   ├── type
+│   ├── provider
+│   ├── uri
+│   ├── username
+│   └── password
+└── transaction-event-stream
+    ├── type
+    ├── connection-count
+    ├── uri
+    ├── certificates
+    └── private-key
+```
+
+In the above example, there are two bindings under the `SERVICE_BINDING_ROOT`
+directory.  The `account-database` and `transaction-event-system` are the names
+of the bindings.  Files within each bindings directory has a special file named
+`type`, and you can rely on the value of that file to identify the type of the
+binding projected into that directory.  In certain directories, you can also see
+another file named `provider`.  The provider is an additional identifier to
+narrow down the type further.  Retrieving the bindings through the bindings
```

Comment:
Through identifying the provider, the intent is to narrow down the type so that the application can identify which backing service should be connected.
For example, let's say the type of binding is `mysql` then the various providers might be ``mariadb`, `oracle`, `bitnami`, `aws-rds`.